synapse/handlers/sync.py

```diff
@@ -485,22 +495,60 @@ def compute_state_delta(self, room_id, batch, sync_config, since_token, now_toke
         # TODO(mjark) Check for new redactions in the state events.
 
         with Measure(self.clock, "compute_state_delta"):
+
+            types = None
+            member_state_ids = {}
```

Comment:
could you add a comment explaining what this is? what does it map from and to?

Reply:
given it's only used in the `full_state` and `batch.limited` branches below, it might make more sense to declare it in those branches.

Reply:
done